### ISSUE TITLE
[Bug]: When `openvino.dll` absolution path contain East Asian character, OpenVINO would not found the device

### ISSUE THREAD

**Issue by xiaoweiChen** [openvinotoolkit/openvino#25715]
### OpenVINO Version

2024.2.0.15519.5c0f38f83f6

### Operating System

Windows System

### Device used for inference

-

### Framework

None

### Issue description

Under English Windows System, if the `openvino.dll` absolution path contain East Asian character(such as, Chinese), The device can not be found.

I tested on my side. (Create the project under `%USERPROFILE%/work` dir)

1. Under Chinese support Windows System, the account use whatever characters, all executable file running well in Visual Studio.
2. Under English Windows System, the account use English characters,"wide-string-demo" can not found device, "narrow-string-demo" can find devices.
3. Under English Windows System, the account use East Asian characters,"wide-string-demo" and "narrow-string-demo" both not found device.

### Step-by-step reproduction

I will make the minimum project to reporduce this issue, I think.
[narraw-wide-string-test-for-ov-device.zip](https://github.com/user-attachments/files/16369149/narraw-wide-string-test-for-ov-device.zip)

With cmake 3.28.0+

```bash
cmake -S<same-with-readme-dir> -B<build-dir>
```

such as:
```bash
cmake -S. -Bbuild # under this code root dir
```

Enter build dir, and open sln with Visual Studio. And make "narrow-string-demo" or "wide-string-demo" as start project and run, then switch start project to another one and run. You can see what I described above.

The demo is very easy to read:
```c++


#include "openvino/openvino.hpp"

#include <vector>
#include <stdexcept>
#include <iostream>

auto ov_inference() {

    ov::Core ie;
    auto avalibable_devices = ie.get_available_devices();

    if (avalibable_devices.empty()) {
        std::string error_message = "No inference device can be used.";
        throw std::runtime_error(error_message);
    }

    std::cout << "The Device:" << std::endl;
    for (auto& device : avalibable_devices) {
        std::cout << "\t" << device << std::endl;
    }
}

int main() {
    try {
        ov_inference();
    }
    catch (const std::exception& ex) {
        std::cerr << ex.what() << std::endl;
    }
}
```

### Relevant log output

>Base on the demo code.
>If find the device(s), the program would print them out one by one.
>Otherwise, the function would throw a exception and catched in main() and print "No inference device can be used."

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by andrei-kochin** [comment]
@peterchen-intel  @sammysun0711 can it  be related to the #24520 ? or is it something different?

**Comment by xiaoweiChen** [comment]
Try the latest release 2024.2, this issue is fixed.

Running same code with 2024.3 and 2024.2

<img width="1138" alt="Screenshot 2024-08-01 061839" src="https://github.com/user-attachments/assets/2aec498b-bb51-442f-8207-32f9ae32192d">
<img width="1174" alt="Screenshot 2024-08-01 061938" src="https://github.com/user-attachments/assets/9b2ee5c5-6710-4c08-8e56-e43a8ada1be0">

I think this issue can be closed. :D

**Comment by zhaixuejun1993** [comment]
@peterchen-intel do we need to also port this fix to 2024.2?
